### ISSUE TITLE
Using a .sql file to create table results in Xcode error "Multiple commands produce ..."

### ISSUE THREAD

**Issue by smithi01** [Lighter-swift/Lighter#27]
Your package looks outstanding. Thank you. I would like to use it to replace GRDB in my current app development. I followed the getting started steps but instead of using an existing database, I added stations.sql to my project.

I get this Xcode error.
Multiple commands produce '/Users/iansmith/Library/Developer/Xcode/DerivedData/TryLighterStations-euddgsmncyclyietjarityspmrwp/Build/Products/Debug-iphonesimulator/TryLighterStations.app/station.sql'

The error also provides this information.

- Target 'TryLighterStations' (project 'TryLighterStations') has copy command from '/Users/iansmith/Documents/Developer/Tutorials, Examples and Experiments/Experiments/TryLighterStations/TryLighterStations/TryLighterStations/station.sql' to '/Users/iansmith/Library/Developer/Xcode/DerivedData/TryLighterStations-euddgsmncyclyietjarityspmrwp/Build/Products/Debug-iphonesimulator/TryLighterStations.app/station.sql'
- Target 'TryLighterStations' (project 'TryLighterStations') has copy command from '/Users/iansmith/Library/Developer/Xcode/DerivedData/TryLighterStations-euddgsmncyclyietjarityspmrwp/SourcePackages/plugins/TryLighterStations.output/TryLighterStations/Enlighter/station.sql' to '/Users/iansmith/Library/Developer/Xcode/DerivedData/TryLighterStations-euddgsmncyclyietjarityspmrwp/Build/Products/Debug-iphonesimulator/TryLighterStations.app/station.sql'

This is my station.sql:
CREATE TABLE station (
    country_code TEXT,
    home_page TEXT,
    icon_url TEXT,
    latitude REAL,
    longitude REAL,
    name TEXT NOT NULL,
    ordered_by_name TEXT NOT NULL,
    location TEXT NOT NULL,
    id TEXT NOT NULL PRIMARY KEY,
    url TEXT NOT NULL,
    genre_list TEXT
);

CREATE INDEX country_index ON station (country_code);
CREATE INDEX name_index ON station (name);
CREATE INDEX ordered_by_name_index ON station (ordered_by_name);

CREATE TABLE station_genre (
    genre TEXT NOT NULL,
    station_id TEXT NOT NULL REFERENCES station (id) ON DELETE CASCADE
)

I am using Xcode Version 15.0.1 (15A507) and running on an ios17 simulator.

Perhaps I am not adding the .sql file to my project correctly. I am at a loss about where I went wrong. I would be most grateful for any guidance you could offer.

**Comment by smithi01** [comment]
[TryLighterStations.zip](https://github.com/Lighter-swift/Lighter/files/13404831/TryLighterStations.zip)

**Comment by helje5** [comment]
I'll have a look

**Comment by smithi01** [comment]
Thanks so much. I just had a thought, which fixed it! I renamed station.sql to stationCreation.sql and all is good! Perhaps the problem is that the database and the table had the same name. I have since renamed it again to stationDB-001.sql and all good,

**Comment by kebbbnnn** [comment]
Had the same issue building my iOS project when I upgraded to Xcode 15. What I did is, from Build Phases, I moved the ".sql" file from Copy Bundle Resources to Compile Sources, then everything went back to normal again.

**Comment by helje5** [comment]
Ah, that makes sense. I wonder whether DB embedding broke w/ that thought (they need to be both compiled and copied) (vs .sql files, which you don't usually want to copy as resources).

**Comment by kebbbnnn** [comment]
I just tested, Xcode 15 broke the db embedding too. I experimented with spm and created a swift package for my db manager then moved all db related sources and files (sql and sqlite3) there and now it works.

**Comment by helje5** [comment]
Yes, that would be the workaround I've would have tried as well. I.e. moving things to a [Local Package](https://developer.apple.com/documentation/xcode/organizing-your-code-with-local-packages) probably makes it work. It was already required when importing SPM packages that used package plugins (e.g. [Northwind.swift](https://github.com/Northwind-swift/NorthwindSQLite.swift)).

**Comment by helje5** [comment]
I'm slightly confused where we are with that and what issues we have. Does the new title contain all issues we have here? Do we need a separate issue for embedding DBs?

**Comment by helje5** [comment]
Xcode 16 has another variant of this
```
error: Multiple commands produce '/../DB.sql'
    note: Target 'x' (project 'x') has copy command from '/.../DB.sql' to '/tmp/DerivedData/.../x.app/Contents/Resources/DB.sql'
    note: Target 'x' (project 'x') has copy command from '/tmp/DerivedData/.../Enlighter/DB.sql' to '/tmp/DerivedData/.../x.app/Contents/Resources/DB.sql'
note: Removed stale file '/tmp/DerivedData/x.build/Script-E886F7D62CAEEB4A00ECFE2A.sh'
```
I think it should be fine to both compile the SQL and to include it as a resource. Maybe there is an issue in the Enlighter plugin here.

**Comment by helje5** [comment]
Aha, I suppose that one got fixed :-)

Enlighter.swift:536
```swift
      // So, in Xcode, if a resource is handled by Enlighter, Xcode itself
      // doesn't copy the resource anymore. Likely a bug.
      // So what we do is copy them ourselves into the plugin dir. They then
      // get bundled properly.
```

**Comment by helje5** [comment]
Fixed for Xcode 16.
